backend/service/github/github.go

```diff
@@ -273,3 +275,21 @@ func (s *svc) GetFile(ctx context.Context, ref *RemoteRef, path string) (*File,
 
 	return f, nil
 }
+
+func (s *svc) CompareCommits(ctx context.Context, ref *RemoteRef, compareSHA string) (scgithubv1.CommitCompareStatus, error) {
+	comp, _, err := s.rest.Repositories.CompareCommits(context.Background(), ref.RepoOwner, ref.RepoName, compareSHA, ref.Ref)
+	if err != nil {
+		return scgithubv1.CommitCompareStatus_UNKNOWN, fmt.Errorf("Could not get compare status for %s and %s. %+v", ref.Ref, compareSHA, err)
+	}
+
+	switch strings.ToUpper(*comp.Status) {
+	case scgithubv1.CommitCompareStatus_BEHIND.String():
+		return scgithubv1.CommitCompareStatus_BEHIND, nil
+	case scgithubv1.CommitCompareStatus_AHEAD.String():
+		return scgithubv1.CommitCompareStatus_AHEAD, nil
+	case scgithubv1.CommitCompareStatus_IDENTICAL.String():
+		return scgithubv1.CommitCompareStatus_IDENTICAL, nil
+	default:
+		return scgithubv1.CommitCompareStatus_UNKNOWN, fmt.Errorf("Unknown status %s", *comp.Status)
+	}
+}
```

Comment:
this can be written more simply as below and will not require updating for any changes to the enum values. also uses `comp.GetStatus()` which is slightly safer than dereferencing the pointer.

```go
	cs, ok := scgithubv1.CommitCompareStatus_value[strings.ToUpper(comp.GetStatus())]
	if !ok {
		return scgithubv1.CommitCompareStatus_UNKNOWN, fmt.Errorf("unknown status %s", comp.GetStatus())
	}
	return scgithubv1.CommitCompareStatus(cs), nil
```